src/layers/XRLayer/xr-layer.js

```diff
@@ -0,0 +1,159 @@
+import GL from '@luma.gl/constants';
+import { Layer, project32 } from '@deck.gl/core';
+import { Model, Geometry } from '@luma.gl/core';
+import vs from './xr-layer-vertex';
+import fs from './xr-layer-fragment';
+
+const defaultProps = {
+  rgbTexture: null,
+};
+
+/*
+ * @class
+ * @param {object} props
+ * @param {number} props.transparentColor - color to interpret transparency to
+ * @param {number} props.tintColor - color bias
+ */
+export default class XRLayer extends Layer {
+  getShaders() {
+    return super.getShaders({ vs, fs, modules: [project32] });
+  }
+
+  initializeState() {
+    const attributeManager = this.getAttributeManager();
+
+    attributeManager.add({
+      positions: {
+        size: 3,
+        type: GL.DOUBLE,
+        fp64: this.use64bitPositions(),
+        update: this.calculatePositions,
+        noAlloc: true,
```

Comment:
Feel free to comment each line of the this! Unless it's something that would be obvious to anyone that had gotten this far?